packages/in_app_purchase/example/lib/main.dart

```diff
@@ -27,53 +27,115 @@ class _MyAppState extends State<MyApp> {
-                    ]);
+                    return Center(
+                      child: Text('Error: ' + snapshot.error.toString()),
+                    );
                   }
-
-                  return ListView(children: snapshot.data);
-                })),
+                  List<String> notFound = snapshot.data.notFoundIDs;
+                  return Column(
+                    children: <Widget>[
+                      Center(child: Text('Products')),
+                      Center(
+                        child:
+                            Text('Not found products: ' + notFound.toString()),
+                      ),
+                      Expanded(
+                        child: ListView(
+                          children: _buildProductList(snapshot.data),
+                        ),
+                      ),
+                    ],
+                  );
+                },
+              ),
+            ),
+          ],
+        ),
       ),
     );
   }
 
-  Future<List<Widget>> buildStorefront() async {
+  Future<List<ListTile>> _buildConnectionCheckTile() async {
     final bool available = await InAppPurchaseConnection.instance.isAvailable();
-    final Widget storeHeader = buildListCard(ListTile(
+    final Widget storeHeader = buildListCard(
+      ListTile(
         leading: Icon(available ? Icons.check : Icons.block),
-        title: Text('The store is ' +
-            (available ? 'available' : 'unavailable') +
-            '.')));
-    final List<Widget> children = <Widget>[storeHeader];
+        title: Text(
+            'The store is ' + (available ? 'available' : 'unavailable') + '.'),
+      ),
+    );
+    final List<ListTile> children = <ListTile>[storeHeader];
 
     if (!available) {
-      children.add(buildListCard(ListTile(
-          title: Text('Not connected',
-              style: TextStyle(color: ThemeData.light().errorColor)),
-          subtitle: const Text(
-              'Unable to connect to the payments processor. Has this app been configured correctly? See the example README for instructions.'))));
-    } else {
       children.add(
-          buildListCard(ListTile(title: const Text('Nothing to see yet.'))));
+        buildListCard(
+          ListTile(
+            title: Text('Not connected',
+                style: TextStyle(color: ThemeData.light().errorColor)),
+            subtitle: const Text(
+                'Unable to connect to the payments processor. Has this app been configured correctly? See the example README for instructions.'),
+          ),
+        ),
+      );
     }
-
     return children;
   }
 
+  List<ListTile> _buildProductList(QueryProductDetailsResponse response) {
+    List<ListTile> productDetailsCards = response.productDetails.map(
+      (ProductDetails productDetails) {
+        return buildListCard(ListTile(
+          title: Text(
+            productDetails.title,
+          ),
+          subtitle: Text(
+            productDetails.description,
+          ),
```

Comment:
nit, optional: Can we display formatted price string here somewhere too? Doesn't really matter where for now, maybe just in parens as part of the title (eg "Product title ($X.XX)").